File: php/testcode/weakhash_005.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakhash_md5_integrity
function weakhash005(BenchmarkRequest $req): BenchmarkResponse {
    $path = $req->param('file');
    $checksum = md5_file($path); // vuln-code-snippet vuln-line php_weakhash_md5_integrity
    return BenchmarkResponse::json(['checksum' => $checksum]);
}
// vuln-code-snippet end php_weakhash_md5_integrity

File: ruby/testcode/weakhash_005.rb
require_relative 'shared'
require 'openssl'

# vuln-code-snippet start ruby_weakhash_openssl_md5
def hash_token(req)
  token = req.param('token')
  hashed = OpenSSL::Digest.new('MD5').hexdigest(token) # vuln-code-snippet vuln-line ruby_weakhash_openssl_md5
  BenchmarkResponse.ok(hashed)
end
# vuln-code-snippet end ruby_weakhash_openssl_md5
